hq/app/schedule/CredentialsReportJob.scala

```diff
@@ -21,22 +21,30 @@ class CredentialsReportJob(enabled: Boolean, cacheService: CacheService)(executi
     } else {
       logger.info(s"Running scheduled job: $description")
 
-      getCredentialsReport().foreach{ case (awsAccount, result) =>
-        result.foreach{ credsReport =>
+      val someValue: List[Either[FailedAttempt, Notification]] = getCredentialsReport().toList.map{ case (awsAccount, eFCreds) =>
+        eFCreds.map{ credsReport =>
           val outdatedKeys = findOldAccessKeys(credsReport)
           val missingMfa = findMissingMfa(credsReport)
           val message = createMessage(outdatedKeys, missingMfa)
-          val email = createNotification(Account(awsAccount.id), message)
-          send(email, topicArn, ???)(executionContext)
-          logger.info(s"Completed scheduled job: $description")
+          createNotification(Account(awsAccount.id), message)
         }
       }
+      someValue.foreach{ result: Either[FailedAttempt, Notification] =>
+        result match {
+          case Left(value) =>
+            logger.info(s"check this issue out security devs! $value")
+          case Right(email) =>
+            send(email, topicArn, ???)(executionContext)
```

Comment:
what is the ??? for?

Reply:
It's a placeholder for the AWS SNS client, which I will add in a subsequent PR. Sorry, I should have noted this PR is a WIP as I try to practice opening smaller PRs!